### PR TITLE
#544 - Optional permissions in RepoConfig

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #544 
Implementing global permissions support requires special handling of absence of permissions configured in repository. This requires to return `Optional<Permissions>` in `RepoConfig.permissions()` method. Logic for extracting permissions part from repo config moved from `YamlPermissions` to `RepoConfig` for that

### CODE REVIEW

File: src/test/java/com/artipie/RpPermissionsTest.java [hunk 1]
```diff
@@ -101,4 +97,17 @@ void adminCanDoAnything() {
...
     }
 
+    /**
+     * Permissions from repo-full-config.yml example file.
+     *
+     * @return Permissions parsed from file.
+     */
+    private YamlPermissions permissions() throws IOException {
+        return new YamlPermissions(
+            Yaml.createYamlInput(new File("src/test/resources/repo-full-config.yml"))
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/RpPermissionsTest.java:107]: @olegmoz let's use `TestResource` from asto to obtain content from config file

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz thanks, one suggestion, take a look please

File: src/test/java/com/artipie/RpPermissionsTest.java (hunk 1 above)
Comment (olegmoz) [reply src/test/java/com/artipie/RpPermissionsTest.java]: @olenagerasimova thanks, corrected

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, good to merge

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#549` is not assigned, can't get performer

**Pull request merged**